### DISCUSSION TITLE
Succses with LG S18EQ (S18EQ.NSK und S18EQ.UL2)

### DISCUSSION THREAD

**Discussion by tach2004** [JanM321/esphome-lg-controller#69] [Show and tell]
"Hello everyone and hello Jan :)"

First of all, many, many thanks for the work and implementation of this brilliant project. Of course, also to the colleagues who helped or from whom some of the information came!

With my **LG S18EQ (S18EQ.NSK and S18EQ.UL2)**without wlan and LGThinkQ, everything works wonderfully! Since I was already somewhat familiar with the ESPs and ESPHome, the implementation was really no problem, and the instructions/description were very understandable!

However, there is one thing I still need to test more thoroughly. The fan control for the vertical louvers only works partially. When I select "Both" in Home Assistant, the louver stops at one position after one cycle, and the setting "Both" sometimes switches back and forth between "Horizontal" or "Vertical." Sometimes, after some time or after repeatedly selecting "Both," it works as it should, with the louvers automatically moving up and down.

I believe I may have seen an issue related to this before. I'll have to dig into it over the weekend. :)

Thanks again, and THANK YOU to those who create such BRILLIANT projects and make them available to the community!
Best regards.

By the way: i have 4 pcb´s left:)

**Comment by JanM321** [comment]
> Thanks again, and THANK YOU to those who create such BRILLIANT projects and make them available to the community!

You're welcome!

> However, there is one thing I still need to test more thoroughly. The fan control for the vertical louvers only works partially. When I select "Both" in Home Assistant, the louver stops at one position after one cycle, and the setting "Both" sometimes switches back and forth between "Horizontal" or "Vertical." Sometimes, after some time or after repeatedly selecting "Both," it works as it should, with the louvers automatically moving up and down.
> 
> I believe I may have seen an issue related to this before. I'll have to dig into it over the weekend. :)

Hm that's weird. I'd have to see the log when this happens to determine what's going on.

> By the way: i have 4 pcb´s left:)

If you want to sell some of them, you could mention it in issue #31 or in the [Home Assistant thread](https://community.home-assistant.io/t/lg-ac-wired-controller-integration-via-esphome-esp32/582954).

**Comment by tach2004** [comment]
Hi and thank you for your answer.
I attached the debug protocol. maybe you find something. I also changed from both to horizontal and back. 
3 things i saw over the weekend.

1. When the unit is off and not disconnected from power it changes all the time from both to horizontal. You see this log on the bottom of the debug file at timestamp 16:50:55 but maybe thats normal and ok?!

2. Although I can select between "off", "vertical", "horizontal", and "both" for the fan louver, and they then automatically move back and forth, I can only select a fixed position for the vertical louver via the Airflow 1 Up/Down position (0-6). The horizontal option doesn’t even appear for me but for sure i have this function.

3. Then there's the known issue as written above where sometimes the "both" setting isn't maintained, and the louver complete one cycle and then stop. This doesn't always happen though. Sometimes they complete a cycle, stop, and then resume after about 10 seconds with the selected setting, e.g., "both". Or it switches to "horizontal" automatically, even though I previously selected "both". 

I also found one more error in the debug log, but it's visible quite far down in the file:

```
[16:31:25][D][lg-controller:946]: received C8.12.C0.08.40.00.05.98.00.00.00.00.D4 (13)  
[16:31:25][E][lg-controller:956]: invalid checksum C8.12.C0.08.40.00.05.98.00.00.00.00.D4 (13)  
[16:31:25][D][lg-controller:946]: received C8.12.C0.08.40.00.05.9A.00.00.00.00.D4 (13)  
[16:31:25][E][lg-controller:1013]: ignoring due to previous error C8.12.C0.08.40.00.05.9A.00.00.00.00.D4 (13)  
```
and:
[16:35:49][E][lg-controller:1395]: discarding incomplete data 12.C0.08.40.00.05.97.00.00.00.00.2B (12)

That's it for now. As I said, these are all minor things. The main functions are running smoothly :) Maybe I'll try at some point to read the energy function or the current power output. Maybe I can figure it out.

Thank you again!

Let me know if you need any more  clarifications!

[debug.txt](https://github.com/user-attachments/files/17015181/debug.txt)

**Comment by JanM321** [comment]
> I can only select a fixed position for the vertical louver via the Airflow 1 Up/Down position (0-6). The horizontal option doesn’t even appear for me but for sure i have this function.

This is a known limitation of the LG wall controllers. My units let me set the horizontal position in the ThinQ app, but the wired controllers just don't support this setting, it's not even sent to the controllers. There are some other cases like this unfortunately (unit display light on/off for example). The Wifi module, wireless remote, and wired controller all have a different set of features that overlap some but not completely.

> I also found one more error in the debug log, but it's visible quite far down in the file:

Maybe an issue with the wiring or something. The checksum byte will catch most issues, but this could cause problems if bits in the message are flipped.

> When the unit is off and not disconnected from power it changes all the time from both to horizontal. You see this log on the bottom of the debug file at timestamp 16:50:55 but maybe thats normal and ok?!

At 16:51:38 the controller sends "both". At 16:52:01 the AC sends "horizontal". I don't know where that change came from because nothing is sent/received between this. Maybe the unit doesn't like sending swing mode vertical when it's turned off? Could also be a corruption issue due to wiring or something but I'd expect that to affect more things beside swing mode.

Another thing you could try is deleting [this line](https://github.com/JanM321/esphome-lg-controller/blob/9964e0ad8316ac971dcfcd284e475792635b8043/esphome/components/lg_controller/lg-controller.h#L1479), line 1479 that says `pending_type_a_settings_change_ = true;`. This is a relatively recent change and maybe setting the vertical vane position is interfering with vertical swing mode. Just a long shot, I don't really know if that's an issue.

**Comment by tach2004** [comment]
Hi

OK, thank you again for the answer and the information.
I'll check it again when I get the chance and maybe replace the cable, although I can rule that out. There's not much to it. Maybe the circuit board from jlcpcb? But since I still have 4 left, I'll just swap them too:) I soldered 2 jst xh plugs together because I had the connection on the main board, but no cable was laid out like yours.

But nonetheless, these aren't really problems for me at the moment. I'm glad that someone made something like this available and I can now control my air conditioning via home assistant and by and large, or 99% of the time, it works:)

So thanks again and I'll let you know when I know anything new
